service/src/event/ordering_task.rs

```diff
@@ -628,12 +628,11 @@ mod test {
     }
 
     async fn insert_10_with_9_undelivered(pool: &SqlitePool) {
-        let insertable = get_n_insertable_events(10).await;
-        let mut init = insertable.first().unwrap().to_owned();
+        let mut insertable = get_n_insertable_events(10).await;
+        let mut init = insertable.remove(0);
```

Comment:
this change is O(n) when previously it was constant time. Not sure it matters but thought I would point it out.